stratum/hal/lib/tdi/tdi_sde_helpers.cc

```diff
@@ -403,15 +403,25 @@ ::util::Status GetAllEntries(
   const ::tdi::Device *device = nullptr;
   ::tdi::DevMgr::getInstance().deviceGet(0, &device);
   const auto flags = ::tdi::Flags(0);
-  uint32 entries = 0;
+  std::unique_ptr<::tdi::TableKey> table_key;
+  std::unique_ptr<::tdi::TableData> table_data;
+  tdi_status_t tdi_status;
+  std::unique_ptr<::tdi::Target> dev_tgt;
+  device->createTarget(&dev_tgt);
+  uint32 entries = 0, actual = 0;
+  auto table_type = static_cast<tdi_table_type_e>(table->tableInfoGet()->tableTypeGet());
+
   if (IsPreallocatedTable(*table)) {
```

Comment:
Repeating my previous comments:

1. Both clauses of this `if` statement are identical. _There is no need for a conditional_. The clauses should be merged. Having two identical clauses is confusing to the reader, who's going to pore over them trying to figure out what the difference is.
2. Now that `table->sizeGet()` works for all table types, there is no need for the `IsPreallocatedTable` function. It should be eliminated.

In addition, I don't think you need the `table_type` variable anymore. IIRC, it was moved to `GetSdeTableType` when `IsPreallocatedTable` and `GetSdeTableType` were refactored out of `GetAllEntries`.